bundles/org.eclipse.ui.forms/src/org/eclipse/ui/forms/widgets/ExpandableComposite.java

```diff
@@ -887,15 +887,47 @@ public int getExpansionStyle() {
 	/**
 	 * Programmatically changes expanded state.
 	 *
+	 * Equivalent to <code>setExpanded(expanded, false)</code>. @see {@link #setExpanded(boolean, boolean)}.
+	 *
 	 * @param expanded
 	 *            the new expanded state
 	 */
 	public void setExpanded(boolean expanded) {
-		internalSetExpanded(expanded);
+		setExpanded(expanded, false);
+	}
+
+	/**
+	 * Programmatically changes expanded state.
+	 *
+	 * @param expanded
+	 *            the new expanded state
+	 * @param fireExpanding
+	 *            if true, listeners will be fired
+	 */
+	public void setExpanded(boolean expanded, boolean fireExpanding) {
+		internalSetExpanded(expanded, fireExpanding);
 		if (toggle != null)
 			toggle.setExpanded(expanded);
 	}
 
+	/**
+	 * Performs the expansion state change for the expandable control.
+	 *
+	 * @param expanded
+	 *            the expansion state
+	 * @param fireExpanding
+	 *            if true, listeners will be fired
+	 */
+	private void internalSetExpanded(boolean expanded, boolean fireExpanding) {
```

Comment:
I don't think it's worth a new method here. Couln't this be inlined in `setExpanded` instead?